embulk-core/src/test/java/org/embulk/spi/json/TestJsonParser.java

```diff
@@ -0,0 +1,180 @@
+            int integer = random.nextInt();
+            testWithParseContexts(newInteger(integer), Integer.toString(integer));
+        }
+    }
+
+    @Test
+    public void testArray()
+            throws Exception
+    {
+        testWithParseContexts(newArray(newInteger(0), newString("embulk")), "[0, \"embulk\"]");
+    }
+
+    @Test
+    public void testObject()
+            throws Exception
+    {
+        testWithParseContexts(newMap(newString("c0"), newString("embulk"), newString("c1"), newInteger(10)), "{\"c0\":\"embulk\",\"c1\":10}");
+        testWithParseContexts(newMap(newString("c0"), newArray(newInteger(0), newString("embulk"))), "{\"c0\":[0,\"embulk\"]}");
+    }
+
+    @Test
+    public void testInvalidJson()
+    {
+        List<String> invalidJsons = Lists.newArrayList("[\"invalid\"", "]", "{\"invalid\"", "}");
+        for (String invalidJson : invalidJsons) {
+            try {
+                testWithSingleParseContext(null, invalidJson);
+                fail();
+            }
+            catch (Throwable t) {
+                assertTrue(t instanceof JsonParseException);
+            }
+            try {
+                testWithStreamParseContext(null, invalidJson);
+                fail();
+            }
+            catch (Throwable t) {
+                assertTrue(t instanceof JsonParseException);
+            }
+        }
+    }
+
+    private void testWithParseContexts(Value expected, String json)
+            throws Exception
+    {
+        testWithSingleParseContext(expected, json);
+        testWithStreamParseContext(expected, json);
+    }
+
+    private void testWithParseContexts(Value expected, String json, Assertion assertion)
+            throws Exception
+    {
+        testWithSingleParseContext(expected, json, assertion);
+        testWithStreamParseContext(expected, json, assertion);
+    }
+
+    private void testWithSingleParseContext(Value expected, String json)
+    {
+        testWithSingleParseContext(expected, json, new Assertion() {
+            @Override
+            public void equal(Value expected, Value actual)
+            {
+                assertEquals(expected, actual);
+            }
+        });
+    }
+
+    private void testWithSingleParseContext(Value expected, String json, Assertion asssertion)
+    {
+        asssertion.equal(expected, new JsonParser().parse(json));
+    }
+
+    private void testWithStreamParseContext(Value expected, String json)
+            throws IOException
+    {
+        testWithStreamParseContext(expected, json, new Assertion() {
+            @Override
+            public void equal(Value expected, Value actual)
+            {
+                assertEquals(expected, actual);
+            }
+        });
+    }
+
+    private void testWithStreamParseContext(Value expected, String json, Assertion assertion)
+            throws IOException
+    {
+        try (InputStream in = new ByteArrayInputStream(json.getBytes());
```

Comment:
`String.getBytes` is invalid because it's platform-dependent behavior. Tests may fail on Windows. I think findbugs told you warning. Check `embulk-standards/build/reports/findbugs/test.html` after running tests.
Related: https://github.com/embulk/embulk/commit/933c612aeb9241b32e77f56e6703cdd09b7f5d92